data/core/macros/amla.cfg

```diff
@@ -12,19 +31,272 @@
+        first_time_only=no
+        [filter]
+            {AMLA_FILTERS}
+            [not]
+                id=$amla_currently_advancing_id
+            [/not]
+        [/filter]
+
+        #--------------------
+        # SKIP IF NOT MAX-LEVEL
+        #--------------------
+        {STORE_UNIT_VAR id=$unit.id advances_to amla_candidate_advances_to}
+        [if] {VARIABLE_CONDITIONAL amla_candidate_advances_to equals ""}
+            [then]
+                #--------------------
+                # AVOID RECURSION
+                #--------------------
+                # modifying a unit re-triggers "pre advance", so avoid infinite recursion
+                {VARIABLE amla_currently_advancing_id $unit.id}
+
+                #--------------------
+                # EXPLAIN AMLAS
+                #--------------------
+                [fire_event]
+                    name=explain_amlas
+                [/fire_event]
+
+                #--------------------
+                # RESET AMLAS
+                #--------------------
+                [remove_object]
+                    object_id=universal_amlas_object
+                    id=$unit.id
+                [/remove_object]
+
+                #--------------------
+                # ADD AMLAS
+                #--------------------
+                [modify_unit]
+                    [filter]
+                        id=$unit.id
+                    [/filter]
+                    [object]
+                        id=universal_amlas_object
+                        [effect]
+                            apply_to=remove_advancement
+                            amlas=amla_default
+                        [/effect]
+                        [effect]
+                            apply_to=new_advancement
+
+                            #--------------------
+                            # HITPOINTS
+                            #--------------------
+                            # +9 hitpoints
+                            [advancement]
+                                id=hitpoints_increase
+                                description= _ "Gain +8 hitpoints."
+                                strict_amla=yes
+                                max_times=-1
+                                [effect]
+                                    apply_to=hitpoints
+                                    increase_total=8
+                                [/effect]
+                                {AMLA_EFFECTS_HEAL_AND_RAISE_XP}
+                            [/advancement]
+
+                            #--------------------
+                            # MELEE DAMAGE
+                            #--------------------
+                            # +4 melee damage
+                            [advancement]
+                                id=melee_increase1
+                                description= _ "Gain +4 melee damage."
+                                strict_amla=yes
+                                max_times=-1
```

Comment:
That seems a bit crazy? For hitpoints ok but for damage??

Reply:
+4 damage only applies to attacks with a single strike. Attacks with 2 strikes get +2 damage, and attacks with >2 strikes only get +1 damage.

Reply:
I rather mean the infinite `max_times`. What damage will Delfador for example do at the end of HttT? I guess you're going to make the campaign shorter but no limit still seems a bit much.

Reply:
The current HttT specifically is one example of a campaign I don't think these AMLAs would work well in. Because it's so long, the player will spend most of the campaign with a fully-maxed-out army - meaning the XP cost of an AMLA is irrelevant.

But in shorter campaigns, my experience is that the XP cost of AMLAs ensures they're not overpowered. For example, in TDG Delfador gets access to a familiar, a Raven > Dark Omen > War Harbinger. I've seen some players invest tons of XP in AMLAs, raising the familiar's damage from 23-1 to 35-1 or more. 

Certainly powerful, but spending 350+ XP on one unit leaves the rest of their army underleveled. Since people had fun doing it and its wasn't overpowered, I saw no reason to stop them.